### PR TITLE
Boolean get_query_argument

### PR DESCRIPTION
**Pull request by guynir42**
We need to inherit `get_query_handler` to deal with situations where the expected output is a boolean, not a string. 

Example: `do_something = self.get_query_argument("doSomething", False)` followed by `if do_something:` will do the following:
- If the argument `doSomething` is not given in the query, return `False`, which is desired behavior. 
- If the argument is given as "True", returns the string `"True"` then the `if` will evaluate it as true. Desired behavior. 
- If the argument is given as "False" it returns the string `"False"` which the `if` takes as true. This is the wrong behavior. 

Instead, we override the `get_query_argument` function to act differently if given a default boolean. In that case, it will interpret any of the following inputs: `['true', 't', 'yes', '1']` (in a _case insensitive_ manner) as true. All other values are interpreted as false.

### CODE REVIEW

File: app/handlers/base.py [hunk 1]
```diff
@@ -392,3 +392,9 @@ def push_notification(self, note, notification_type='info'):
             action='baselayer/SHOW_NOTIFICATION',
             payload={'note': note, 'type': notification_type},
         )
+
+    def get_query_argument(self, name, default=RequestHandler._ARG_DEFAULT, strip=True):
```
Comment (stefanv) [inline app/handlers/base.py:396]:
```suggestion
    def get_query_argument(self, *args, **kwargs):
```

File: app/handlers/base.py [hunk 2]
```diff
@@ -392,3 +392,9 @@ def push_notification(self, note, notification_type='info'):
...
+        ret = RequestHandler.get_query_argument(self, name, default, strip)
```
Comment (stefanv) [inline app/handlers/base.py:397]:
```suggestion
        arg = super().get_query_argument(*args, **kwargs)
```

File: app/handlers/base.py [hunk 3]
```diff
@@ -392,3 +392,9 @@ def push_notification(self, note, notification_type='info'):
...
+        if type(default) == bool:
```
Comment (stefanv) [inline app/handlers/base.py:398]:
```suggestion
        if type(kwargs.get('default', None)) == bool:
```

File: app/handlers/base.py [hunk 4]
```diff
@@ -392,3 +392,9 @@ def push_notification(self, note, notification_type='info'):
...
+            ret = type(ret) == str and ret.lower() in ['true', 'yes', 't', '1']
```
Comment (stefanv) [inline app/handlers/base.py:399]:
```suggestion
            return arg.lower() in ['true', 'yes', 't', '1']
```

File: app/handlers/base.py [hunk 5]
```diff
@@ -392,3 +392,9 @@ def push_notification(self, note, notification_type='info'):
...
+        return ret
```
Comment (stefanv) [inline app/handlers/base.py:400]:
```suggestion
```

File: app/handlers/base.py (hunk 4 above)
Comment (guynir42) [reply app/handlers/base.py:399]: What if `arg` is None?

File: app/handlers/base.py (hunk 5 above)
Comment (guynir42) [reply app/handlers/base.py:400]: I think we need to return after the `if` or we won't get the arguments for the non-boolean case.

File: app/handlers/base.py (hunk 4 above)
Comment (stefanv) [reply app/handlers/base.py:399]: Yes, that can happen. How about `str(arg).lower() in ...`.

**Comment by guynir42** [comment]
@stefanv ready for re-review.

**Comment by stefanv** [comment]
It's a bit scary that we don't have tests for baselayer :/  Fortunately, it's well tested throug SkyPortal.
Thanks @guy.

**Pull request merged**